### PR TITLE
Remove uses of `--full-dart-sdk` from the engine_v2 builders.

### PR DESCRIPTION
**Pull request by eyebrowsoffire**
This updates some of the engine_v2 builders to get them away from the old "host build to make the web sdk" pattern.

### CODE REVIEW

File: ci/builders/linux_host_engine.json [hunk 1]
```diff
@@ -78,7 +78,6 @@
                         "out/host_debug/zip_archives/linux-x64/font-subset.zip",
                         "out/host_debug/zip_archives/flutter_patched_sdk.zip",
                         "out/host_debug/zip_archives/dart-sdk-linux-x64.zip",
-                        "out/host_debug/zip_archives/flutter-web-sdk-linux-x64.zip",
```
Comment (godofredoc) [inline ci/builders/linux_host_engine.json:81]: Does this mean we don't need this artifact anymore?

Comment (eyebrowsoffire) [reply ci/builders/linux_host_engine.json:81]: That's correct. The flutter tool no longer uses any of the host specific flutter-web-sdk artifacts (i.e. `flutter-web-sdk-{platform-name}.zip`). It now just uses `flutter-web-sdk.zip` which is host-agnostic, and that is produced by the linux_host_engine builder.

Comment (godofredoc) [reply ci/builders/linux_host_engine.json:81]: Thanks for the explanation. LGTM.

**Review by godofredoc — APPROVED** [review]

File: ci/builders/linux_host_engine.json (hunk 1 above)
Comment (eyebrowsoffire) [reply ci/builders/linux_host_engine.json:81]: Thank you!

**Pull request merged**